rir/src/compiler/util/cfg.cpp

```diff
@@ -200,7 +200,7 @@ const DominanceGraph::BBList& DominanceGraph::dominators(BB* bb) const {
 
 void DominanceGraph::dominatorTreeNext(
     BB* bb, const std::function<void(BB*)>& apply) const {
-    BreadthFirstVisitor::run(bb, [&](BB* b) {
```

Comment:
this change is 100% wrong.

if we have
```
  A
 / \
B   C
 \ /
  D
```

then starting from A we want to visit in the order B,C,D. Which you discover by enumerating dominated nodes in breadth-first order of the cfg. This has nothing to do with the traversal order of the dominator tree, which is
```
A
|\ \
B C D
```